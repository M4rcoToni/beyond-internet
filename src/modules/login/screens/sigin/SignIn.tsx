import { Content, Logo, Title } from './styles'

import { useNavigation } from '@react-navigation/native'
import { Button, Container, Input, SubTitle, Link } from '@shared/components'
import { User } from 'databases/modules/users/model/User'
import { getUserByCPF } from 'databases/modules/users/repository/UserRepository'

export function SignIn() {
  const navigation = useNavigation()
  async function handleSignIn() {
    const callback = (user: User | boolean) => {
      if (user !== false) console.log(user)
      else console.log('Usuário não encontrado')
    }

    getUserByCPF('047.196.130-2', callback)
  }
  return (
    <Container>
      <Title>Além da Internet</Title>
      <Content>
        <SubTitle>Entre com seu CPF e Senha</SubTitle>
        <Input placeholder="CPF" />
        <Input placeholder="Senha" />
        <Button title="Entrar" type="PRIMARY" onPress={handleSignIn} />
        <Link
          title="Cadastre-se"
          onPress={() => navigation.navigate('signUp')}
        />
      </Content>
      <Logo source={require('../../../shared/assets/logo_ifsul.png')} />
    </Container>
  )
}
