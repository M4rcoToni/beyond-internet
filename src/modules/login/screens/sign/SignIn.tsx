import { useNavigation } from '@react-navigation/native'
import { yupResolver } from '@hookform/resolvers/yup'
import Toast from 'react-native-toast-message'
import { Controller, useForm } from 'react-hook-form'

import { CPFMask } from '@shared/helpers/CpfMask'
import { useAuth } from '@shared/hooks/useAuth'
import {
  LoginFormDataProps,
  signInSchema,
} from '@modules/login/utils/FormValidator'
import { Button, Container, Input, SubTitle, Link } from '@shared/components'
import { Content } from './styles'

export function SignIn() {
  const navigation = useNavigation()
  const { signIn } = useAuth()

  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<LoginFormDataProps>({
    resolver: yupResolver(signInSchema),
  })

  async function handleSignIn({ cpf, password }: LoginFormDataProps) {
    try {
      await signIn(cpf, password)
    } catch (error) {
      Toast.show({
        type: 'error',
        text1:
          error instanceof Error ? error.message : 'Erro ao realizar login',
      })
    }
  }

  return (
    <Container>
      {/* <Title>Além da Internet</Title>ß */}
      <Content>
        <SubTitle text="Entre com seu CPF e Senha" />

        <Controller
          control={control}
          name="cpf"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Informe o CPF"
              onChangeText={(text) => {
                const maskedText = CPFMask(text)
                onChange(maskedText)
              }}
              maxLength={14}
              value={value}
              errorMessage={errors.cpf?.message}
            />
          )}
        />

        <Controller
          control={control}
          name="password"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Senha"
              onChangeText={onChange}
              value={value}
              secureTextEntry
              errorMessage={errors.password?.message}
            />
          )}
        />

        <Button
          title="Entrar"
          type="PRIMARY"
          style={{ height: 56 }}
          onPress={handleSubmit(handleSignIn)}
        />
        <Link
          title="Cadastre-se"
          onPress={() => navigation.navigate('signUp')}
        />
      </Content>
      {/* <Logo source={require('../../../shared/assets/logo_ifsul.png')} /> */}
      <Toast />
    </Container>
  )
}
