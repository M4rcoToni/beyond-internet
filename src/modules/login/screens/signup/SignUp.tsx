import React from 'react'
import { useNavigation } from '@react-navigation/native'

import { Button, Container, Input, SubTitle, Link } from '@shared/components'
import { useForm, Controller } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'

import { Content, Logo } from './styles'
import { CPFMask } from '@shared/helpers/CpfMask'
import { useAuth } from '@shared/hooks/useAuth'
import { FormDataProps, signUpSchema } from '@modules/login/utils/FormValidator'

export function SignUp() {
  const navigation = useNavigation()
  const { signUp } = useAuth()
  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<FormDataProps>({
    resolver: yupResolver(signUpSchema),
  })

  async function handleSignUp({
    cpf,
    name,
    password,
    passwordConfirm,
  }: FormDataProps) {
    signUp({ cpf, name, password, passwordConfirm })
  }

  return (
    <Container>
      {/* <Title>Al√©m da Internet</Title> */}
      <Content>
        <SubTitle>Realize seu cadastro</SubTitle>
        <Controller
          control={control}
          name="name"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Nome Completo"
              onChangeText={onChange}
              value={value}
              errorMessage={errors.name?.message}
            />
          )}
        />
        <Controller
          control={control}
          name="cpf"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Informe o CPF"
              onChangeText={(text) => {
                const maskedText = CPFMask(text)
                onChange(maskedText)
              }}
              maxLength={14}
              value={value}
              errorMessage={errors.cpf?.message}
            />
          )}
        />

        <Controller
          control={control}
          name="password"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Senha"
              onChangeText={onChange}
              value={value}
              errorMessage={errors.password?.message}
            />
          )}
        />

        <Controller
          control={control}
          name="passwordConfirm"
          render={({ field: { onChange, value } }) => (
            <Input
              placeholder="Confirme sua senha"
              onChangeText={onChange}
              value={value}
              errorMessage={errors.passwordConfirm?.message}
            />
          )}
        />

        <Button
          title="Cadastre-se"
          type="PRIMARY"
          onPress={handleSubmit(handleSignUp)}
        />
        <Link title="Volar para o login" onPress={() => navigation.goBack()} />
      </Content>
      <Logo source={require('../../../shared/assets/logo_ifsul.png')} />
    </Container>
  )
}
