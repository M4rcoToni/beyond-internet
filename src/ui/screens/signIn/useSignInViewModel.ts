import { useNavigation } from '@react-navigation/native'
import Toast from 'react-native-toast-message'
import { LoginFormDataProps } from '@data/utils/FormValidator'
import { Result } from '@data/result'
import { AuthRepository } from '@data/repositories/auth'
import { useState } from 'react'
import { useAuth } from '@data/contexts/AuthContext'

export function useSignInViewModel(authRepository: AuthRepository) {
  const { setUserData } = useAuth()
  const navigation = useNavigation()
  const [loading, setLoading] = useState(false)

  async function handleSignIn({ cpf, password }: LoginFormDataProps) {
    setLoading(true)
    try {
      const res = await authRepository.login(cpf, password)
      const user = res?.getValue()
      Toast.show({
        type: 'success',
        text1: 'Login efetuado com sucesso',
      })
      if (user) {
        setTimeout(() => {
          setUserData(user)
        }, 1300)
      }
    } catch (error) {
      Toast.show({
        type: 'error',
        text1: error instanceof Result ? error.getError()?.message : 'Erro',
      })
    } finally {
      setLoading(false)
    }
  }

  return {
    navigation,
    handleSignIn,
    loading,
  }
}
