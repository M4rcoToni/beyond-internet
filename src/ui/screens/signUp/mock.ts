// jest.mock('../foo-bar-baz', () => {
//   const originalModule = jest.requireActual('../foo-bar-baz');
//
//   //Mock the default export and named export 'foo'
//   return {
//     __esModule: true,
//     ...originalModule,
//     default: jest.fn(() => 'mocked baz'),
//     foo: 'mocked foo',
//   };
// });

const mockCreateUser = jest
  .fn()
  .mockImplementation(async ({ cpf, name, password }) => {
    return {
      name,
      cpf,
      password,
    }
  })

jest.mock('../../../data/repositories/auth/index', () => {
  const originalModule = jest.requireActual(
    '../../../data/repositories/auth/index',
  )

  return {
    __esModule: true,
    ...originalModule,
    AuthRepository: jest.fn().mockImplementation(() => {
      return {
        createUser: mockCreateUser,
      }
    }),
  }
})

export { mockCreateUser }
