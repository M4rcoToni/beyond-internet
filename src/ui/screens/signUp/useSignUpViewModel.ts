import { useNavigation } from '@react-navigation/native'
import Toast from 'react-native-toast-message'
import { FormDataProps, LoginFormDataProps } from '@data/utils/FormValidator'
import { Result } from '@data/result'
import { AuthRepository } from '@data/repositories/auth'
import { useState } from 'react'
import { useAuth } from '@data/contexts/AuthContext'

export function useSignUpViewModel(authRepository: AuthRepository) {
  const navigation = useNavigation()
  const [loading, setLoading] = useState(false)
  const { setUserData } = useAuth()

  async function handleSignUp({
    cpf,
    name,
    password,
    passwordConfirm,
  }: FormDataProps) {
    setLoading(true)
    try {
      if (password !== passwordConfirm) {
        throw new Result(false, undefined, new Error('Senhas não conferem'))
      }

      const user = await authRepository.createUser({
        cpf,
        name,
        password,
      })

      Toast.show({
        type: 'success',
        text1: 'Usuário criado com sucesso',
      })

      if (user) {
        setTimeout(() => {
          setUserData(user)
        }, 600)
      }
    } catch (error) {
      Toast.show({
        type: 'error',
        text1: error instanceof Result ? error.getError()?.message : 'Erro',
      })
    } finally {
      setLoading(false)
    }
  }

  return {
    navigation,
    handleSignUp,
    loading,
  }
}
