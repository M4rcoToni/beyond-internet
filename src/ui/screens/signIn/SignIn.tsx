import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { yupResolver } from '@hookform/resolvers/yup'
import Toast from 'react-native-toast-message'
import { Controller, useForm } from 'react-hook-form'
import { Image } from 'expo-image'
import { Content } from './styles'
import { Button, Container, Input, Link, SubTitle } from '@ui/components'
import { LoginFormDataProps, signInSchema } from 'data/utils/FormValidator'
import { ScrollView } from 'react-native'
import { useSignInViewModel } from './useSignInViewModel'
import { AuthService } from '@data/services/auth'
import { AuthRepository } from '@data/repositories/auth'
import { CPFMask } from '@data/utils/CpfMask'

export function SignIn() {
  const { navigate } = useNavigation()

  const { handleSignIn, loading } = useSignInViewModel(
    new AuthRepository(new AuthService()),
  )

  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<LoginFormDataProps>({
    resolver: yupResolver(signInSchema),
  })

  return (
    <Container>
      <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
        <Content>
          <Image
            source={require('../../assets/AppIcon.png')}
            style={{ width: 120, height: 100 }}
            transition={600}
            responsivePolicy="live"
            alt="Logo do IFSUL"
          />
          <SubTitle text="Entre com seu CPF e Senha" />

          <Controller
            control={control}
            name="cpf"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Informe o CPF"
                onChangeText={(text) => {
                  const maskedText = CPFMask(text)
                  onChange(maskedText)
                }}
                maxLength={14}
                value={value}
                keyboardType="numeric"
                errorMessage={errors.cpf?.message}
                testID="cpf-input"
              />
            )}
          />

          <Controller
            control={control}
            name="password"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Senha"
                onChangeText={onChange}
                value={value}
                secureTextEntry
                errorMessage={errors.password?.message}
                testID="sigin-password-input"
              />
            )}
          />

          <Button
            title="Entrar"
            type="PRIMARY"
            isLoading={loading}
            style={{ height: 56 }}
            onPress={handleSubmit(handleSignIn)}
            testID="signin-button"
          />
          <Link
            title="Cadastre-se"
            testID="signup-link"
            onPress={() => navigate('signUp')}
          />
        </Content>
      </ScrollView>

      <Toast />
    </Container>
  )
}
