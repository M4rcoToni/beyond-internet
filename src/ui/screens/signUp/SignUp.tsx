import React from 'react'
import { useNavigation } from '@react-navigation/native'

import { Controller, useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import Toast from 'react-native-toast-message'

import { Content } from './styles'
import { Image } from 'expo-image'

import { Button, Container, Input, Link, SubTitle } from '@ui/components'
import { FormDataProps, signUpSchema } from '@data/utils/FormValidator'
import { ScrollView } from 'react-native'
import { CPFMask } from '@data/utils/CpfMask'
import { useSignUpViewModel } from './useSignUpViewModel'
import { AuthRepository } from '@data/repositories/auth'
import { AuthService } from '@data/services/auth'

export function SignUp() {
  const navigation = useNavigation()
  const { handleSignUp } = useSignUpViewModel(
    new AuthRepository(new AuthService()),
  )

  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<FormDataProps>({
    resolver: yupResolver(signUpSchema),
  })

  return (
    <Container>
      {/* <Title>Al√©m da Internet</Title> */}
      <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
        <Content>
          <Image
            source={require('../../assets/AppIcon.png')}
            style={{ width: 120, height: 100 }}
            transition={600}
            responsivePolicy="live"
            alt="Logo do IFSUL"
          />
          <SubTitle text="Cadastro" testID="sign-up-title" />
          <Controller
            control={control}
            name="name"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Nome Completo"
                onChangeText={onChange}
                value={value}
                errorMessage={errors.name?.message}
                testID="name-input"
              />
            )}
          />
          <Controller
            control={control}
            name="cpf"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Informe seu CPF"
                onChangeText={(text) => {
                  const maskedText = CPFMask(text)
                  onChange(maskedText)
                }}
                maxLength={14}
                value={value}
                errorMessage={errors.cpf?.message}
                testID="cpf-input"
              />
            )}
          />

          <Controller
            control={control}
            name="password"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Senha"
                onChangeText={onChange}
                value={value}
                errorMessage={errors.password?.message}
                testID="password-input"
              />
            )}
          />

          <Controller
            control={control}
            name="passwordConfirm"
            render={({ field: { onChange, value } }) => (
              <Input
                placeholder="Confirme sua senha"
                onChangeText={onChange}
                value={value}
                errorMessage={errors.passwordConfirm?.message}
                testID="password-confirm-input"
              />
            )}
          />

          <Button
            title="Cadastre-se"
            type="PRIMARY"
            style={{ height: 56 }}
            onPress={handleSubmit(handleSignUp)}
            testID="signup-button"
          />
          <Link
            title="Voltar para o login"
            onPress={() => navigation.goBack()}
          />
        </Content>
      </ScrollView>
      <Toast />
    </Container>
  )
}
