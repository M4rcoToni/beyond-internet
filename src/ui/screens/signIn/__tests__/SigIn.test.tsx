import React from 'react'
import { SignIn } from '@screens/signIn/SignIn'
import { fireEvent, render, screen } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import { SignUp } from '@screens/signUp/SignUp'

describe('SignIn', () => {
  it('render screen', () => {
    render(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>,
    )

    expect(screen).toMatchSnapshot()
  })

  it('should type cpf on input', () => {
    render(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>,
    )

    const input = screen.getByTestId('cpf-input')

    fireEvent.changeText(input, '047.196.130-20')
    expect(input.props.value).toBe('047.196.130-20')
  })

  it('should type password on input', () => {
    render(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>,
    )

    const input = screen.getByTestId('sigin-password-input')

    fireEvent.changeText(input, '123456')

    expect(input.props.value).toBe('123456')
  })

  it('should show error message when cpf is invalid', async () => {
    render(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>,
    )

    const input = screen.getByTestId('cpf-input')
    const button = screen.getByTestId('signin-button')

    fireEvent.changeText(input, '123')
    fireEvent.press(button)

    expect(await screen.findByText('CPF inválido')).toBeTruthy()
  })

  it('should show error message when password is short', async () => {
    render(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>,
    )

    const input = screen.getByTestId('sigin-password-input')
    const button = screen.getByTestId('signin-button')

    fireEvent.changeText(input, '123')

    fireEvent.press(button)

    expect(
      await screen.findByText('A senha deve ter no mínimo 6 dígitos.'),
    ).toBeTruthy()
  })

  it('should navigate to SignUp screen', async () => {
    render(
      <NavigationContainer>
        <SignIn />
        <SignUp />
      </NavigationContainer>,
    )

    const button = screen.getByTestId('signup-button')

    fireEvent.press(button)

    expect(await screen.findAllByText('Cadastro')).toBeTruthy()
  })
})
