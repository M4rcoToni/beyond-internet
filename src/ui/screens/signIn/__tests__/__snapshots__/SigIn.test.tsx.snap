// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn render screen 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#f2f2f2",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 14,
            "justifyContent": "center",
          }
        }
      >
        <ViewManagerAdapter_ExpoImage
          accessibilityLabel="Logo do IFSUL"
          contentFit="cover"
          contentPosition={
            {
              "left": "50%",
              "top": "50%",
            }
          }
          height={100}
          onError={[Function]}
          onLoad={[Function]}
          onLoadStart={[Function]}
          onProgress={[Function]}
          placeholder={[]}
          responsivePolicy="live"
          source={[]}
          style={
            {
              "height": 100,
              "width": 120,
            }
          }
          transition={
            {
              "duration": 600,
            }
          }
          width={120}
        />
        <Text
          size={16}
          style={
            {
              "color": "#000000",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Entre com seu CPF e Senha
        </Text>
        <TextInput
          keyboardType="numeric"
          maxLength={14}
          onChangeText={[Function]}
          placeholder="Informe o CPF"
          placeholderTextColor="#FFFFFF"
          style={
            {
              "backgroundColor": "#7C7C8A",
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
              "color": "#FFFFFF",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 16,
              "maxHeight": 56,
              "maxWidth": 400,
              "minHeight": 56,
              "paddingLeft": 16,
              "width": "100%",
            }
          }
          testID="cpf-input"
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Senha"
          placeholderTextColor="#FFFFFF"
          secureTextEntry={true}
          style={
            {
              "backgroundColor": "#7C7C8A",
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
              "color": "#FFFFFF",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 16,
              "maxHeight": 56,
              "maxWidth": 400,
              "minHeight": 56,
              "paddingLeft": 16,
              "width": "100%",
            }
          }
          testID="sigin-password-input"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#00875F",
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
              "height": 56,
              "justifyContent": "center",
              "maxHeight": 56,
              "maxWidth": 400,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="signin-button"
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontFamily": "Poppins_700Bold",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Entrar
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Poppins_400Regular",
              "fontSize": 16,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="signup-link"
        >
          <Text
            size={16}
            style={
              {
                "color": "#000000",
                "fontFamily": "Poppins_400Regular",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Cadastre-se
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
    testID="toastAnimatedContainer"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFF",
          "borderLeftColor": "#69C779",
          "borderLeftWidth": 5,
          "borderRadius": 6,
          "elevation": 2,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
          "width": 340,
        }
      }
      testID="toastTouchableContainer"
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 25,
            },
            undefined,
          ]
        }
        testID="toastContentContainer"
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
